src/main/java/org/opensearch/knn/common/exception/DeleteModelWhenInTrainStateException.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.common.exception;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.logging.LoggerMessageFormat;
+import org.opensearch.rest.RestStatus;
+
+public class DeleteModelWhenInTrainStateException extends OpenSearchException {
+    public DeleteModelWhenInTrainStateException(String msg, Object... args) {
+        super(LoggerMessageFormat.format(msg, args));
+    }
+
+    @Override
+    public RestStatus status() {
+        return RestStatus.CONFLICT;
```

Comment:
Can you pls tell why are we overriding it and what does CONFLICT do here ?

Reply:
The issue is that the resource cannot be deleted because it is not in the proper state. This aligns with:
```
     * The request could not be completed due to a conflict with the current state of the resource. This code is
     * only allowed in situations where it is expected that the user might be able to resolve the conflict and
     * resubmit the request. The response body SHOULD include enough information for the user to recognize the
     * source of the conflict. Ideally, the response entity would include enough information for the user or user
     * agent to fix the problem; however, that might not be possible and is not required.
```
From https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/RestStatus.java#L354-L366.

I can add a comment on it.